analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Argument.cs

```diff
@@ -44,6 +44,11 @@ private static ProgramState ProcessArgument(ProgramState state, IArgumentOperati
         {
             state = state.SetSymbolConstraint(notNullSymbol, ObjectConstraint.NotNull);
         }
+        if (argument.WrappedOperation.TrackedSymbol(state) is { } symbol
+            && state[symbol] is { } symbolValue)
+        {
+            state = state.SetSymbolValue(symbol, symbolValue.WithoutConstraint<CollectionConstraint>());
+        }
```

Comment:
I was thinking to put this in `CollectionTracker` to keep the collection logic bundled. Wdyt?